### PR TITLE
Enable refreshing buckets with actual values from Sentinel Hub.

### PR DESCRIPTION
**Pull request by Ardweaden**
Fetch actual number of available processing units and requests every `REFRESH_BUCKETS_SEC` seconds and update the buckets-

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,2 +1,3 @@
 CLIENT_ID=
-CLIENT_SECRET="..."
\ No newline at end of file
+CLIENT_SECRET="..."
+REFRESH_BUCKETS_SEC=
```
Comment (zcernigoj) [inline .env.example:3]: Add to Readme.md what does this do.

File: syncer/syncer.py [hunk 2]
```diff
@@ -193,19 +243,18 @@ def fill_bucket(policy_id, fill_interval_s, fill_quantity, capacity, scheduled_a
...
-    CLIENT_ID = os.environ.get("CLIENT_ID")
-    CLIENT_SECRET = os.environ.get("CLIENT_SECRET")
-    # Docker-compose doesn't strip double quotes when reading from .env; however running this file from
-    # command line decodes the secret incorrectly if the quotes are absent. To avoid having two different
-    # ways of writing .env files, we remove the quotes here if present:
-    if CLIENT_SECRET.startswith('"') and CLIENT_SECRET.endswith('"'):
-        CLIENT_SECRET = CLIENT_SECRET[1:-1]
-    if not CLIENT_ID or not CLIENT_SECRET:
-        raise Exception("Please supply CLIENT_ID and CLIENT_SECRET env vars!")
+    REFRESH_BUCKETS_SEC = int(os.environ.get("REFRESH_BUCKETS_SEC", 500))
```
Comment (zcernigoj) [inline syncer/syncer.py]: This fails if `REFRESH_BUCKETS_SEC` env var is not set (running with `docker-compose up`).

```bash
syncer    | Traceback (most recent call last):
syncer    |   File "syncer.py", line 281, in <module>
syncer    |     main()
syncer    |   File "syncer.py", line 257, in main
syncer    |     REFRESH_BUCKETS_SEC = int(os.environ.get("REFRESH_BUCKETS_SEC", 500))
syncer    | ValueError: invalid literal for int() with base 10: ''
syncer exited with code 1
```

File: syncer/syncer.py [hunk 3]
```diff
@@ -176,6 +202,30 @@ def fill_bucket(policy_id, fill_interval_s, fill_quantity, capacity, scheduled_a
...
+        try:
+            if auth_token is None or will_auth_token_soon_expire(auth_token):
+                auth_token = request_auth_token(CLIENT_ID, CLIENT_SECRET)
+
+            user_id = extract_user_id(auth_token)
+            stats = fetch_current_stats(auth_token, user_id)
+        except Exception as ex:
+            logging.warning(f"Refreshing buckets failed! {str(ex)}")
+
+        now = time.time()
```
Comment (zcernigoj) [inline syncer/syncer.py]: This isn't used so it can probably be removed.

**Review by zcernigoj — CHANGES_REQUESTED** [review]
In general it seems to work as it should.

File: .env.example (hunk 1 above)
Comment (Ardweaden) [reply .env.example:3]: Added.

File: syncer/syncer.py (hunk 2 above)
Comment (Ardweaden) [reply syncer/syncer.py]: Thanks, I didn't realise docker-compose automatically sets it to a blank string if not available. Having a default was wrong anyway.

File: .env.example (hunk 1 above)
Comment (zcernigoj) [reply .env.example:3]: Thanks. Fixed.

File: syncer/syncer.py (hunk 2 above)
Comment (zcernigoj) [reply syncer/syncer.py]: Works now.

File: syncer/syncer.py (hunk 3 above)
Comment (zcernigoj) [reply syncer/syncer.py]: was removed

**Review by zcernigoj — APPROVED** [review]

**Pull request merged**